File: modules/printer-drivers/ios/services/bluetooth/BluetoothEventHandler.swift
import Foundation

protocol BluetoothEventHandler: AnyObject {
    func onDeviceConnected(deviceName: String, deviceAddress: String)
    func onDeviceDisconnected()
    func onConnectionFailed(error: String)
    func onConnectionLost()
    func onDataReceived(data: Data)
    func onBluetoothStateChanged(isAvailable: Bool, isEnabled: Bool)
}

File: modules/printer-drivers/android/src/main/java/expo/modules/printerdrivers/services/bluetooth/BluetoothEventHandler.kt
package expo.modules.printerdrivers.services.bluetooth

interface BluetoothEventHandler {
    fun onDeviceConnected(deviceName: String, deviceAddress: String)
    fun onDeviceDisconnected()
    fun onConnectionFailed(error: String)
    fun onConnectionLost()
    fun onDataReceived(data: ByteArray)
}